VSAN disk claiming fails to add all but last Capacity disk

SUMMARY:  **Multiple "`-d device -d device -d device`" flags are ignored, only 1 Capacity disk is built by govc**

Hi All - I've been working fairly extensively with `govc` over the last few weeks to automate vCenter deployment and VSAN stroage cluster creation.  I'm having a heck of a time with VSAN.  When I run the appropriate commands from the ESXi host itself to `claim` disks - I can properly tag and claim and define Cache/Capacity disks.  Here's an example on the ESXi host:

```
esxcli vsan storage tag add -d naa.55cd2e4152138838 -t capacityFlash
esxcli vsan storage tag add -d naa.55cd2e4151c605c5 -t capacityFlash
esxcli vsan storage tag add -d naa.55cd2e414e326509 -t capacityFlash
esxcli vsan storage add -s naa.55cd2e414e3264b0 -d naa.55cd2e4152138838 -d naa.55cd2e4151c605c5 -d naa.55cd2e414e326509
```

Checking the cluster membership - it shows this working.
```
# esxcli vsan storage list | egrep "^naa|Group UUID"
naa.55cd2e414e3264b0
   VSAN Disk Group UUID: 521d74b2-c4fb-37b0-1171-ea0099e9cbf8
naa.55cd2e4152138838
   VSAN Disk Group UUID: 521d74b2-c4fb-37b0-1171-ea0099e9cbf8
naa.55cd2e414e326509
   VSAN Disk Group UUID: 521d74b2-c4fb-37b0-1171-ea0099e9cbf8
naa.55cd2e4151c605c5
   VSAN Disk Group UUID: 521d74b2-c4fb-37b0-1171-ea0099e9cbf8
```

Now I remove the disks from VSAN, and clear the capacityFlash tag.

Running the EXACT equivalent `govc` commands, from a remote system:
```
govc host.esxcli vsan storage tag add -d naa.55cd2e4152138838 -t capacityFlash
govc host.esxcli vsan storage tag add -d naa.55cd2e4151c605c5 -t capacityFlash
govc host.esxcli vsan storage tag add -d naa.55cd2e414e326509 -t capacityFlash
govc host.esxcli vsan storage add -s naa.55cd2e414e3264b0 -d naa.55cd2e4152138838 -d naa.55cd2e4151c605c5 -d naa.55cd2e414e326509
```

Nets **_only 2 of the disks added_** to the VSAN cluster - the Cache, and very last Capacity disk:

```
# govc host.esxcli vsan storage list | egrep "^Device|^VSANDiskGroupUUID"
Device:                        naa.55cd2e414e326509
VSANDiskGroupUUID:             523346d8-7e21-35d8-5d3b-e463b167d8ff
Device:                        naa.55cd2e414e3264b0
VSANDiskGroupUUID:             523346d8-7e21-35d8-5d3b-e463b167d8ff
```

If I switch the ordering up - it appears that the Cache disk (`-s naa...`) and the very last Capacity disk (`-d naa...`) are the only two disks that govc is operating on.

**_The first and second listed Capacity disks are not properly operated on and being added to the VSAN cluster._**

It would (on the surface) appear that the command line flag parsing is not correctly picking up all 3 of the Capacity disks that I am specifying.  Is there something different in how I'm supposed to pass the multiple `-d device -d device -d device` flags for GoVC to correctly handle them ?? 

NOTE - the `esxcli vsan storage add ...` command appears to only be capable of being run once - as it MUST include a Cache and Capacity disks - so you can not (as far as I have figured out) add new disks to an existing VSAN cluster via that command path ... 

Added the `debug` directory from the `govc host.esxcli vsan storage add ....` command - in case the request/response/headers are helpful?

[govc-vsan-capacity-claim-failure-debug.zip](https://github.com/vmware/govmomi/files/5011583/govc-vsan-capacity-claim-failure-debug.zip)

@dougm - thank you - much appreciated that you're working on this - truly !   Unfortunately, I can't get a clean build of the `govc` binary.  I did a clean check out of `dougm/govmomi`, _git checkout_ the `issues-2071` branch, and ran make ... which results in fails in `tests`:

```
...snip...
?   	github.com/vmware/govmomi/vslm/methods	[no test files]
?   	github.com/vmware/govmomi/vslm/types	[no test files]
FAIL
make: *** [go-test] Error 1
```

I'm not very familiar with the project code/build layout ... though do have some (very minor) experience with building Golang projects and managing them (digitalrebar/provision)...  My apologies on the extra overhead on this - but if you have any pointers on how to get a Linux amd65 govc binary built out of this ... I'm all ears...

--- a/govc/host/esxcli/command.go
+++ b/govc/host/esxcli/command.go
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/vmware/govmomi/govc/flags"
 	"github.com/vmware/govmomi/internal"
 )
 
@@ -93,18 +94,18 @@ func (c *Command) Moid() string {
 // Parse generates a flag.FlagSet based on the given []CommandInfoParam and
 // returns arguments for use with methods.ExecuteSoap
 func (c *Command) Parse(params []CommandInfoParam) ([]internal.ReflectManagedMethodExecuterSoapArgument, error) {
-	flags := flag.NewFlagSet(strings.Join(c.name, " "), flag.ExitOnError)
-	vals := make([]string, len(params))
+	fs := flag.NewFlagSet(strings.Join(c.name, " "), flag.ExitOnError)
+	vals := make([]flags.StringList, len(params))
 
 	for i, p := range params {
 		v := &vals[i]
 		for _, a := range p.Aliases {
 			a = strings.TrimPrefix(a[1:], "-")
-			flags.StringVar(v, a, "", p.Help)
+			fs.Var(v, a, p.Help)
 		}
 	}
 
-	err := flags.Parse(c.args)
+	err := fs.Parse(c.args)
 	if err != nil {
 		return nil, err
 	}
@@ -112,19 +113,22 @@ func (c *Command) Parse(params []CommandInfoParam) ([]internal.ReflectManagedMet
 	args := []internal.ReflectManagedMethodExecuterSoapArgument{}
 
 	for i, p := range params {
-		if vals[i] == "" {
-			continue
+		if len(vals[i]) != 0 {
+			args = append(args, c.Argument(p.Name, vals[i]...))
 		}
-		args = append(args, c.Argument(p.Name, vals[i]))
 	}
 
 	return args, nil
 }
 
-func (c *Command) Argument(name string, val string) internal.ReflectManagedMethodExecuterSoapArgument {
+func (c *Command) Argument(name string, args ...string) internal.ReflectManagedMethodExecuterSoapArgument {
+	var vars []string
+	for _, arg := range args {
+		vars = append(vars, fmt.Sprintf("<%s>%s</%s>", name, arg, name))
+	}
 	return internal.ReflectManagedMethodExecuterSoapArgument{
 		Name: name,
-		Val:  fmt.Sprintf("<%s>%s</%s>", name, val, name),
+		Val:  strings.Join(vars, ""),
 	}
 }
 

